File: update_footers.js
const fs = require('fs');
const path = require('path');

const servicesPath = path.join(__dirname, 'services.html');
const indexPath = path.join(__dirname, 'index.html');
const projectsPath = path.join(__dirname, 'projects.html');
const learnPath = path.join(__dirname, 'learn.html');
const blogPath = path.join(__dirname, 'blog.html');
const privacyPath = path.join(__dirname, 'privacy-policy.html');

// 1. Read services.html
let servicesHtml = fs.readFileSync(servicesPath, 'utf8');

// Extract footer
const footerMatch = servicesHtml.match(/<footer[\s\S]*?<\/footer>/);
if (!footerMatch) {
    console.error("Could not find footer in services.html");
    process.exit(1);
}
let footerHtml = footerMatch[0];

// Refactor footer contact info
// Replace calling phone number with the new one from the screenshot
footerHtml = footerHtml.replace(/0547787867/g, '+971 052 777 3352');
// Replace email address with the one from the screenshot
footerHtml = footerHtml.replace(/info@smartzoneuae.com/g, 'info@arthouseinterior.co');
footerHtml = footerHtml.replace(/mailto:info@smartzoneuae.com/g, 'mailto:info@arthouseinterior.co');

console.log("Extracted and refactored footer HTML.");

// List of target files that need the footer added/updated
const targets = [
    { name: 'projects.html', path: projectsPath },
    { name: 'learn.html', path: learnPath },
    { name: 'blog.html', path: blogPath }
];

targets.forEach(target => {
    let html = fs.readFileSync(target.path, 'utf8');
    
    // Check if footer already exists
    if (html.includes('<footer') || html.includes('</footer')) {
        console.log(`${target.name} already contains a footer. Replacing it...`);
        html = html.replace(/<footer[\s\S]*?<\/footer>/, footerHtml);
    } else {
        // Find insert position (before FOOTER SCRIPTS comment)
        const insertMatch = html.match(/<!--\s*={5,}\s*FOOTER SCRIPTS\s*={5,}\s*-->/) || 
                            html.match(/<!--\s*FOOTER SCRIPTS\s*-->/) ||
                            html.match(/<!--\s*={5,}\s*[\s\S]*?FOOTER SCRIPTS[\s\S]*?={5,}\s*-->/);
        
        if (insertMatch) {
            console.log(`Inserting footer into ${target.name} before comment: "${insertMatch[0].replace(/\n/g, ' ')}"`);
            const index = html.indexOf(insertMatch[0]);
            html = html.substring(0, index) + footerHtml + '\n\n' + html.substring(index);
        } else {
            console.warn(`Could not find footer scripts comment in ${target.name}. Appending before first script tag...`);
            // Fallback: insert before first script tag at the bottom
            const index = html.lastIndexOf('<script');
            if (index !== -1) {
                html = html.substring(0, index) + footerHtml + '\n\n' + html.substring(index);
            } else {
                console.error(`Could not find insert point in ${target.name}`);
            }
        }
    }
    
    // Make sure contact details inside the rest of the page (in case there's any) are also updated
    html = html.replace(/info@smartzoneuae.com/g, 'info@arthouseinterior.co');
    html = html.replace(/mailto:info@smartzoneuae.com/g, 'mailto:info@arthouseinterior.co');
    
    fs.writeFileSync(target.path, html, 'utf8');
    console.log(`Updated ${target.name}`);
});

// Update index.html and services.html footers
let indexHtml = fs.readFileSync(indexPath, 'utf8');
indexHtml = indexHtml.replace(/<footer[\s\S]*?<\/footer>/, footerHtml);
indexHtml = indexHtml.replace(/info@smartzoneuae.com/g, 'info@arthouseinterior.co');
indexHtml = indexHtml.replace(/mailto:info@smartzoneuae.com/g, 'mailto:info@arthouseinterior.co');
fs.writeFileSync(indexPath, indexHtml, 'utf8');
console.log("Updated index.html");

servicesHtml = servicesHtml.replace(/<footer[\s\S]*?<\/footer>/, footerHtml);
servicesHtml = servicesHtml.replace(/info@smartzoneuae.com/g, 'info@arthouseinterior.co');
servicesHtml = servicesHtml.replace(/mailto:info@smartzoneuae.com/g, 'mailto:info@arthouseinterior.co');
fs.writeFileSync(servicesPath, servicesHtml, 'utf8');
console.log("Updated services.html");

// Update privacy-policy.html email
let privacyHtml = fs.readFileSync(privacyPath, 'utf8');
privacyHtml = privacyHtml.replace(/info@smartzoneuae.com/g, 'info@arthouseinterior.co');
privacyHtml = privacyHtml.replace(/mailto:info@smartzoneuae.com/g, 'mailto:info@arthouseinterior.co');
fs.writeFileSync(privacyPath, privacyHtml, 'utf8');
console.log("Updated privacy-policy.html");

console.log("All files updated successfully!");

File: update_footers.py
import os
import re

dir_path = os.path.dirname(os.path.abspath(__file__))

services_path = os.path.join(dir_path, 'services.html')
index_path = os.path.join(dir_path, 'index.html')
projects_path = os.path.join(dir_path, 'projects.html')
learn_path = os.path.join(dir_path, 'learn.html')
blog_path = os.path.join(dir_path, 'blog.html')
privacy_path = os.path.join(dir_path, 'privacy-policy.html')

# Read services.html
with open(services_path, 'r', encoding='utf-8') as f:
    services_html = f.read()

# Extract footer
footer_match = re.search(r'<footer[\s\S]*?</footer>', services_html)
if not footer_match:
    print("Could not find footer in services.html")
    exit(1)
footer_html = footer_match.group(0)

# Refactor footer info:
# Replace call number with the screenshot call number
footer_html = footer_html.replace('0547787867', '+971 052 777 3352')
# Replace email address with the screenshot email address
footer_html = footer_html.replace('info@smartzoneuae.com', 'info@arthouseinterior.co')

print("Extracted and refactored footer HTML.")

targets = [
    ('projects.html', projects_path),
    ('learn.html', learn_path),
    ('blog.html', blog_path)
]

for name, path in targets:
    with open(path, 'r', encoding='utf-8') as f:
        html = f.read()
    
    if '<footer' in html or '</footer' in html:
        print(f"{name} already contains a footer. Replacing it...")
        html = re.sub(r'<footer[\s\S]*?</footer>', footer_html, html)
    else:
        # Find insert position (before FOOTER SCRIPTS comment)
        insert_match = (
            re.search(r'<!--\s*={5,}\s*FOOTER SCRIPTS\s*={5,}\s*-->', html) or
            re.search(r'<!--\s*FOOTER SCRIPTS\s*-->', html) or
            re.search(r'<!--\s*={5,}\s*[\s\S]*?FOOTER SCRIPTS[\s\S]*?={5,}\s*-->', html)
        )
        
        if insert_match:
            print(f"Inserting footer into {name} before comment.")
            index = html.find(insert_match.group(0))
            html = html[:index] + footer_html + '\n\n' + html[index:]
        else:
            print(f"Could not find comment in {name}. Finding last script tag.")
            index = html.rfind('<script')
            if index != -1:
                html = html[:index] + footer_html + '\n\n' + html[index:]
            else:
                print(f"Could not find insert point in {name}")
                
    # Update emails in target file
    html = html.replace('info@smartzoneuae.com', 'info@arthouseinterior.co')
    
    with open(path, 'w', encoding='utf-8') as f:
        f.write(html)
    print(f"Updated {name}")

# Update index.html and services.html
with open(index_path, 'r', encoding='utf-8') as f:
    index_html = f.read()
index_html = re.sub(r'<footer[\s\S]*?</footer>', footer_html, index_html)
index_html = index_html.replace('info@smartzoneuae.com', 'info@arthouseinterior.co')
with open(index_path, 'w', encoding='utf-8') as f:
    f.write(index_html)
print("Updated index.html")

services_html = re.sub(r'<footer[\s\S]*?</footer>', footer_html, services_html)
services_html = services_html.replace('info@smartzoneuae.com', 'info@arthouseinterior.co')
with open(services_path, 'w', encoding='utf-8') as f:
    f.write(services_html)
print("Updated services.html")

# Update privacy-policy.html
with open(privacy_path, 'r', encoding='utf-8') as f:
    privacy_html = f.read()
privacy_html = privacy_html.replace('info@smartzoneuae.com', 'info@arthouseinterior.co')
with open(privacy_path, 'w', encoding='utf-8') as f:
    f.write(privacy_html)
print("Updated privacy-policy.html")

print("All files updated successfully!")
